odk/src/main/java/org/motechproject/odk/service/AbstractFormDefinitionImportService.java

```diff
@@ -0,0 +1,173 @@
+            XformParser parser = new XformParserFactory().getParser(configuration.getType());
+
+            for (String def : xmlFormDefinitions) {
+                formDefinitions.add(parser.parse(def, configuration.getName()));
+            }
+            return formDefinitions;
+        } catch (XPathExpressionException e) {
+            throw new FormDefinitionImportException(e);
+        }
+    }
+
+    /**
+     * Hook method that may be overridden by an implementing subclass. Makes an HTTP request to the external
+     * application and returns a list of URLs for the form definitions.
+     * @param configuration {@link Configuration}
+     * @return A list of strings containing the URLs for each form definition.
+     * @throws FormDefinitionImportException If HTTP request fails or returns a malformed XML list of URLs.
+     */
+    protected List<String> getFormUrls(Configuration configuration) throws FormDefinitionImportException {
+        HttpGet request = new HttpGet(configuration.getUrl() + FORM_LIST_PATH);
+        try {
+            HttpResponse response = client.execute(request);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            return parseToUrlList(responseBody);
+
+        } catch (IOException|ParseUrlException e) {
+            throw new FormDefinitionImportException(e);
+        }
+    }
+
+    /**
+     * Hook method that may be overriden by an implementing subclass. For each URL in the list of
+     * URLs, this makes an HTTP request to the external application to fetch an XML form definition. This returns
+     * a list of XML form definitions.
+     * @param formUrls A list of URLs for each form definition.
+     * @param configuration {@link Configuration}
+     * @return A list of XML form definitions represented as a list of strings.
+     * @throws IOException If any of the HTTP requests fail.
+     */
+    protected List<String> getXmlFormDefinitions(List<String> formUrls, Configuration configuration) throws IOException {
+        List<String> formDefinitions = new ArrayList<>();
+
+        for (String url : formUrls) {
+            HttpGet request = new HttpGet(url);
+            HttpResponse response = client.execute(request);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            formDefinitions.add(responseBody);
+        }
+        return formDefinitions;
+    }
+
+    protected Header generateBasicAuthHeader(HttpUriRequest request, Configuration configuration) {
+        Header basicAuthHeader;
+        BasicScheme basicScheme = new BasicScheme();
+
+        try {
+            basicAuthHeader = basicScheme.authenticate(
+                    new UsernamePasswordCredentials(configuration.getUsername(), configuration.getPassword()),
+                    request,
+                    HttpClientContext.create());
+        } catch (Exception e) {
+            return null;
```

Comment:
same issue as before, this can will either lead to a library error in the best case. In the worst case auth gets silently ignored and it's left to the user to figure this out.